pywbemtools/pywbemcli/_cmd_listener.py

```diff
@@ -0,0 +1,473 @@
+
+    If such a listener process terminates, the corresponding listener no longer
+    exists, so there is no notion of a stopped listener nor does a listener
+    have an operational status. The `pywbemcli listener stop` command terminates
+    a listener gracefully.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'pywbemcli --help') can also be specified before the
+    'connection' keyword.
+    """
+    pass  # pylint: disable=unnecessary-pass
+
+
+@listener_group.command('run', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_run(context, **options):
+    """
+    Run as a named WBEM indication listener.
+
+    Run this command as a named WBEM indication listener until it gets
+    terminated, e.g. by a keyboard interrupt, break signal (e.g. kill), or the
+    `pywbemcli listener stop` command.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener run lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_run(context, options))
+
+
+@listener_group.command('start', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(LISTEN_OPTIONS)
+@add_options(help_option)
+@click.pass_obj
+def listener_start(context, **options):
+    """
+    Start a named WBEM indication listener in the background.
+
+    A listener with that name must not be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener start lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_start(context, options))
+
+
+@listener_group.command('stop', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(help_option)
+@click.pass_obj
+def listener_stop(context, name):
+    """
+    Stop a named WBEM indication listener.
+
+    The listener is stopped gracefully.
+
+    A listener with that name must be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener stop lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_stop(context, name))
+
+
+@listener_group.command('show', cls=PywbemcliCommand,
+                        options_metavar=CMD_OPTS_TXT)
+@click.argument('name', type=str, metavar='NAME', required=True)
+@add_options(help_option)
+@click.pass_obj
+def listener_show(context, name):
+    """
+    Show a named WBEM indication listener.
+
+    A listener with that name must be running, otherwise the command fails.
+
+    Examples:
+
+      pywbemcli listener stop lis1
+    """
+    context.execute_cmd(lambda: cmd_listener_show(context, name))
+
+
+@listener_group.command('list', cls=PywbemcliCommand,
```

Comment:
I had thought about combining list and show (i.e. list with a GLOB for names to show). However, I assume that we will want to present more information about number of indications received, etc. and that can be done through show so combining is an open questions.

Reply:
Right now, they both show the same information because it is short enough. Once we add properties like what action is taken, where is the output logged, etc, it may become different.

Reply:
OK as it stands.